crates/schemalate/src/firebolt/firebolt_schema_builder.rs

```diff
@@ -0,0 +1,406 @@
+use std::iter::FromIterator;
+
+use super::errors::*;
+use super::firebolt_queries::{CreateTable, DropTable, InsertFromTable};
+use super::firebolt_types::{Column, FireboltType, Table, TableSchema, TableType};
+
+use json::schema::types;
+use protocol::flow::MaterializationSpec;
+use protocol::flow::{inference::Exists, materialization_spec::Binding};
+use serde::{Deserialize, Serialize};
+
+pub const FAKE_BUNDLE_URL: &str = "https://fake-bundle-schema.estuary.io";
+
+#[derive(Serialize, PartialEq, Debug)]
+pub struct BindingBundle {
+    pub create_table: String,
+    pub create_external_table: String,
+    pub drop_table: String,
+    pub drop_external_table: String,
+    pub insert_from_table: String,
+}
+
+#[derive(Serialize, PartialEq, Debug)]
+pub struct FireboltQueriesBundle {
+    bindings: Vec<BindingBundle>,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct EndpointConfig {
+    aws_key_id: String,
+    aws_secret_key: String,
+    s3_bucket: String,
+    s3_prefix: String,
+}
+
+#[derive(Deserialize)]
+pub struct Resource {
+    pub table: String,
+    pub table_type: String,
+}
+
+pub fn build_firebolt_schema(binding: &Binding) -> Result<TableSchema, Error> {
+    // TODO / question: is it safe to assume these are available when the driver sends them?
```

Comment:
Yes, IMO given the presumed context of the Flow runtime + wrapping connector sending this in, we can assume it's always present. If it's somehow not, it's an error of implementation (as opposed to an error of user input) and crashing is the best way of handling it.